File: 04.CLIESC/CLIESC_TicketPremium_SOAP_GR08/src/main/java/ec/edu/monster/model/Factura.java
package ec.edu.monster.model;

import java.util.Date;

public class Factura {
    private long idFactura;
    private String idCliente;
    private Date fecha;
    private double subtotal;
    private double iva;
    private double total;
    private int estado; // 1: pagado, 0: pendiente

    public Factura() {}

    public long getIdFactura() { return idFactura; }
    public void setIdFactura(long idFactura) { this.idFactura = idFactura; }
    public String getIdCliente() { return idCliente; }
    public void setIdCliente(String idCliente) { this.idCliente = idCliente; }
    public Date getFecha() { return fecha; }
    public void setFecha(Date fecha) { this.fecha = fecha; }
    public double getSubtotal() { return subtotal; }
    public void setSubtotal(double subtotal) { this.subtotal = subtotal; }
    public double getIva() { return iva; }
    public void setIva(double iva) { this.iva = iva; }
    public double getTotal() { return total; }
    public void setTotal(double total) { this.total = total; }
    public int getEstado() { return estado; }
    public void setEstado(int estado) { this.estado = estado; }
}

File: 04.CLIMOV/app/src/main/java/ec/edu/monster/model/Factura.kt
package ec.edu.monster.model
import java.util.Date

data class Factura(
    var idFactura: Long = 0,
    var idCliente: String = "",
    var vendedor: String = "",
    var subtotal: Double = 0.0,
    var iva: Double = 0.0,
    var total: Double = 0.0,
    var estado: Int = 0,
    var fechaEmision: Date? = null
)
